src/file_layer.ts

```diff
@@ -37,11 +40,31 @@ type EMSFormats = EmsFileLayerFormatGeoJson | EmsFileLayerFormatTopoJson;
 export class FileLayer extends AbstractEmsService {
   protected readonly _config: FileLayerConfig;
 
+  private _getVectorDataOfType = _.memoize(
```

Comment:
I wonder if we should push the caching into the instance of the EMSClient instead, rather than have a memoized result on each individual FileLayer object. The reason for this is that this cache, in theory, can grow with however many layers EMS-exposes. E.g. if a client just loops over all the `FileLayer` objects and gets the geojson, this will keep 60+ files in memory.

I'd consider using a single LRU-cache (https://www.npmjs.com/package/lru-cache), maybe with a size of 10, and expose it on the EMSClient.

e.g. 

`EMSClient#getGeoJsonFromCache(layerId)`

and

`EMSClient#cacheGeoJson(layerId)`

`FileLayer#getGeoJson` can then just use these methods.

Reply:
Good idea. I was wondering about that, too.

Reply:
Maybe this is over engineering, but what about using local storage for caching, so the state is saved between EMS client life cycles? A simple expiration logic would be enough since our assets don't change much.

I believe there are localstorage polyfills for node users and our own tests, but never used them though.

Reply:
> Maybe this is over engineering, but what about using local storage for caching, so the state is saved between EMS client life cycles? A simple expiration logic would be enough since our assets don't change much.
> 
> I believe there are localstorage polyfills for node users and our own tests, but never used them though.

I think it's better to retain a limited cache in memory rather than in the local storage. We may need to fix data in the service and I don't think we can invalidate user's locally stored data easily. 

If we really wanted to have persistent caching, I think we would use [Service Workers](https://developers.google.com/web/fundamentals/primers/service-workers#cache_and_return_requests) in the browser. There we can set TTLs and have native cross-browser support. But I don't think we use Service Workers anywhere in Kibana now and I don't want to add them without having a larger discussion with the team.

Reply:
Fair enough 👍